pgml-apps/chatbot/pgml_chatbot/main.py

```diff
@@ -0,0 +1,262 @@
+import asyncio
+from pgml import Database
+import logging
+from rich.logging import RichHandler
+from rich.progress import track
+from rich import print
+import os
+from dotenv import load_dotenv
+import glob
+import argparse
+from time import time
+import openai
+import signal
+import json
+import ast
+
+def handler(signum, frame):
+    print('Exiting...')
+    exit(0)
+
+signal.signal(signal.SIGINT, handler)
+
+parser = argparse.ArgumentParser(description="Process some integers.")
```

Comment:
Have you ever tried Click? https://pypi.org/project/click/

Reply:
Yeah.. click is my go to! It doesn't support async.